src/test/java/jenkins/plugins/git/GitBranchSCMHeadTest.java

```diff
@@ -0,0 +1,73 @@
+package jenkins.plugins.git;
+
+import hudson.FilePath;
+import hudson.Functions;
+import hudson.model.Queue;
+import org.apache.commons.io.FileUtils;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.recipes.LocalData;
+
+import java.io.File;
+import java.net.URL;
+
+import static org.junit.Assert.*;
+import static org.junit.Assume.assumeFalse;
+
+public class GitBranchSCMHeadTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule() {
+        @Override
+        public void before() throws Throwable {
+            if (!Functions.isWindows() && "testMigrationNoBuildStorm".equals(this.getTestDescription().getMethodName())) {
+                URL res = getClass().getResource("/jenkins/plugins/git/GitBranchSCMHeadTest/testMigrationNoBuildStorm_repositories.zip");
+                final File path = new File("/tmp/JENKINS-48061");
```

Comment:
This hardcoded path ("/tmp/JENKINS-48061") embedded in the test creates the same types of problems as the [JenkinsRule hardcoded path](https://github.com/jenkinsci/jenkins-test-harness/blob/2407c7762a4a4885f6c9537e4dc71b23bacdb072/src/main/java/org/jvnet/hudson/test/TemporaryDirectoryAllocator.java#L53).

If different users are running Jenkins unit tests on the same Unix based computer, only one of them will be allowed to run this test successfully because other users won't have permission to remove the test directory.

See my pull request for a multi-platform proposal that worked in my simple tests.

Could we add an "after" which removes the directory with the hard coded name?

Reply:
Yea, I usually don't like this way of doing it, but I couldn't figure out how to have a dynamic path without doing a configuration change and thereby voiding the test.

Reply:
Would a relative path (`tmp/JENKINS-48061`), instead of an absolute path (`/tmp/JENKINS-48061`), be of any help? Just thinking out loud.

An `@After` like @MarkEWaite suggested would be cool, too.

Reply:
ideally I'd add a "lock file" that contains a time-stamp and if the time-stamp is less than 5 minutes old then sleep until the "lock-file" is deleted... if more than 5 minutes old, wipe the temp dir.

But there should be some way to template in the path without needing to hardcode

Reply:
Did we get a solution to this? If not, file a JIRA, leave a TODO comment and we can move on

Reply:
I added an `@After` to clean up after itself.